### PR TITLE
Linting and tests/refactoring for resource and signature factory

### PR DESCRIPTION
**Pull request by spirosikmd**

### CODE REVIEW

**Comment by spirosikmd** [comment]
Currently `handleOnEnd` and `handleOnData` functions have a log of side effects. This should be refactored after version and after tests are in place.

File: src/signing.js [hunk 1]
```diff
@@ -101,66 +95,93 @@ class SignatureFactory {
...
+  sign() {
+    this.headers['Authorization'] = this.authHeader();
+
+    return {
+      headers: this.headers,
+      url: (this.queryParameters) ? `${this.url}?${this.queryParameters}` : this.url
+    };
+  }
+
+  static getDateTime() {
```
Comment (tzengerink) [inline src/signing.js]: I like it when the static methods defined before the constructor, instead of on the bottom of the class. What do you think?

Comment (spirosikmd) [inline src/signing.js]: I don't really have a strong preference. I kind of prefer having them at the end. We should anyway have a consistent way of doing it. Is it ok to leave it like this for now?

Comment (tzengerink) [inline src/signing.js]: Sure it is! Like you say it's just a preference.

**Pull request merged**